pkg/loadbalancer/experimental/benchmark/benchmark.go

```diff
@@ -346,3 +352,243 @@ func ServicesAndSlices(testSize int) (svcs []*slim_corev1.Service, epSlices []*k
+				}
+				for wantAddr := range epSlices[i].Backends { // There is only one element in this map.
+					if fe.Backends[0].AddrCluster != wantAddr {
+						err = errors.Join(err, fmt.Errorf("Incorrect backend address for frontend #%06d, got %v, want %v", i, fe.Backends[0].AddrCluster, wantAddr))
+					}
+				}
+
+				i++
+			}
+		}
+	}
+
+	{
+		if backendsNo := writer.Backends().NumObjects(txn); backendsNo != len(epSlices) {
+			err = errors.Join(err, fmt.Errorf("Incorrect number of backends, got %d, want %d", backendsNo, len(epSlices)))
+		} else {
+			i := 0
+			for be := range writer.Backends().All(txn) {
+				want := epSlices[i]
+				for wantAddr, wantBe := range want.Backends { // There is only one element in this map.
+					if be.AddrCluster != wantAddr {
+						err = errors.Join(err, fmt.Errorf("Incorrect address for backend #%06d, got %v, want %v", i, be.AddrCluster, wantAddr))
+					}
+					for _, wantPort := range wantBe.Ports { // There is only one element in this map.
+						if be.Port != wantPort.Port {
+							err = errors.Join(err, fmt.Errorf("Incorrect port for backend #%06d, got %v, want %v", i, be.Port, wantPort.Port))
+						}
+						if be.Protocol != wantPort.Protocol {
+							err = errors.Join(err, fmt.Errorf("Incorrect protocol for backend #%06d, got %v, want %v", i, be.Protocol, wantPort.Protocol))
+						}
+					}
+					if be.NodeName != wantBe.NodeName {
+						err = errors.Join(err, fmt.Errorf("Incorrect node name for backend #%06d, got %v, want %v", i, be.NodeName, wantBe.NodeName))
+					}
+				}
+				if be.Instances.Len() != 1 {
+					err = errors.Join(err, fmt.Errorf("Incorrect instances count for backend #%06d, got %v, want %v", i, be.Instances.Len(), 1))
+				} else {
+					for svcName, instance := range be.Instances.All() { // There should
+						if svcName.Name != svcs[i].Name {
+							err = errors.Join(err, fmt.Errorf("Incorrect service name for backend #%06d, got %v, want %v", i, svcName.Name, svcs[i].Name))
+						}
+						if state, tmpErr := instance.State.String(); tmpErr != nil || state != "active" {
+							err = errors.Join(err, fmt.Errorf("Incorrect state for backend #%06d, got %q, want %q", i, state, "active"))
+						}
+						if instance.PortName != svcs[i].Spec.Ports[0].Name {
+							err = errors.Join(err, fmt.Errorf("Incorrect instance port name for backend #%06d, got %q, want %q", i, instance.PortName, svcs[i].Spec.Ports[0].Name))
+						}
+					}
+				}
+
+				i++
+			}
+		}
+	}
+
+	return err
+}
+
+var (
+	nodePortAddrs = []netip.Addr{
+		netip.MustParseAddr("10.0.0.3"),
+		netip.MustParseAddr("2002::1"),
+	}
+)
+
+func testHive(maps experimental.LBMaps,
+	services chan resource.Event[*slim_corev1.Service],
+	pods chan resource.Event[*slim_corev1.Pod],
+	endpoints chan resource.Event[*k8s.Endpoints],
+	writer **experimental.Writer,
+	db **statedb.DB,
+	bo **experimental.BPFOps,
```

Comment:
I understand this C style of doing things and why in this case. But its very un-golang like, I feel this deserves some inline comments for the uninitiated.